diff -up ATLAS/config.c.broken ATLAS/config.c
--- ATLAS/config.c.broken	2008-02-13 17:33:21.000000000 -0500
+++ ATLAS/config.c	2008-02-13 17:34:03.000000000 -0500
@@ -3254,7 +3254,7 @@ enum MACHTYPE GetArch(FILE *fpout, FILE 
          fprintf(stdout, "   1. IA-64 Itanium 1\n");
          fprintf(stdout, "   2. IA-64 Itanium 2\n");
          fprintf(stdout, "   3. Other/UNKNOWN\n");
-         i = GetIntRange(3, 1, 2, "", "machine number");
+         i = GetIntRange(2, 1, 2, "", "machine number");
          if (i == 1) mach = IA64Itan;
          else if (i == 2) mach = IA64Itan2;
          break;
