diff -up wrk/CONFIG/src/probe_comp.c.wrk wrk/CONFIG/src/probe_comp.c
--- wrk/CONFIG/src/probe_comp.c.wrk	2013-09-17 12:53:26.863023304 +0200
+++ wrk/CONFIG/src/probe_comp.c	2013-09-17 15:39:22.967912120 +0200
@@ -1059,8 +1059,10 @@ void GetComps(enum OSTYPE OS, enum MACHT
  * If ptrbits is set to manual override, add -m32/64 to gnu compilers
  * but not on Itaniums or Apple's munged gcc 3 compiler!
  */
-   if (ptrbits && arch != IA64Itan && arch != IA64Itan2)
+	   printf("TU SOOOOOOOOOOOM vonku\n");
+   if (ptrbits && arch != IA64Itan && arch != IA64Itan2 && arch != ARMv7)
    {
+	   printf("TU SOOOOOOOOOOOM\n");
       for (i=0; i < NCOMP; i++)
       {
          sp = GetPtrbitsFlag(OS, arch, ptrbits, usrcomps[i]);
diff -up wrk/CONFIG/src/SpewMakeInc.c.wrk wrk/CONFIG/src/SpewMakeInc.c
--- wrk/CONFIG/src/SpewMakeInc.c.wrk	2013-09-17 15:51:55.991929088 +0200
+++ wrk/CONFIG/src/SpewMakeInc.c	2013-09-17 15:52:01.738952113 +0200
@@ -386,6 +386,7 @@ char *GetPtrbitsFlag(enum OSTYPE OS, enu
  * RETURNS: string forcing setting of ptrbits for gcc
  */
 {
+	printf("XXXXXX\n");
    char *sp = "";
    int i, j, k;
 
